db-service/lib/SQLService.js

```diff
@@ -199,7 +199,32 @@ class SQLService extends DatabaseService {
     // REVISIT: It's not yet 100 % clear under which circumstances we can rely on db constraints
     return (super.onDELETE = /* cds.env.features.assert_integrity === 'db' ? this.onSIMPLE : */ deep_delete)
     async function deep_delete(/** @type {Request} */ req) {
-      let { compositions } = req.target
+      const transitions = getTransition(req.query.target, this)
+      if (transitions.target !== transitions.queryTarget) {
+        const elements = transitions.queryTarget.keys
+          ? Object.keys(transitions.queryTarget.keys)
+          : Object.keys(transitions.queryTarget.elements).filter(
+              key => !transitions.queryTarget.elements[key].isAssociation,
+            )
+        const matchedKeys = elements
+          .filter(key => key !== 'IsActiveEntity' && transitions.mapping.has(key))
```

Comment:
`key !== 'IsActiveEntity'` doesn't look right. It should probable be a check that the element is `virtual`.